src/Pelagos/Bundle/AppBundle/Controller/UI/RemotelyHostedDatasetsController.php

```diff
@@ -28,4 +36,102 @@ public function defaultAction()
         $GLOBALS['pelagos']['title'] = 'Remotely Hosted Datasets';
         return $this->render('PelagosAppBundle:List:RemotelyHostedDatasets.html.twig');
     }
+
+    /**
+     * Mark as Remotely Hosted Dataset.
+     *
+     * @param Request $request The Symfony request object.
+     *
+     * @Route("/{udi}")
+     *
+     * @Method("POST")
+     *
+     * @return TerminateResponse A response.
+     */
+    public function postAction(Request $request)
+    {
+        // Checks authorization of users
+        if (!$this->isGranted('ROLE_DATA_REPOSITORY_MANAGER')) {
+            return $this->render('PelagosAppBundle:template:AdminOnly.html.twig');
+        }
+
+        $udi = $request->attributes->get('udi');
+        $datasets = $this->entityHandler->getBy(Dataset::class, array('udi' => $udi));
+
+        if (!empty($datasets)) {
+            $dataset = $datasets[0];
+            $datasetSubmission = $dataset->getDatasetSubmission();
+            $datasetStatus = $dataset->getDatasetStatus();
+
+            if ($datasetStatus === Dataset::DATASET_STATUS_ACCEPTED) {
+                if (DatasetSubmission::TRANSFER_STATUS_REMOTELY_HOSTED !== $datasetSubmission->getDatasetFileTransferStatus()) {
+                    $datasetSubmission->setDatasetFileTransferStatus(DatasetSubmission::TRANSFER_STATUS_REMOTELY_HOSTED);
+
+                    $actor = $this->get('security.token_storage')->getToken()->getUser()->getUserId();
```

Comment:
Cant we get the user using $this->getUser()->getPerson()?

Reply:
No. I copied from DataRestrictionController

Reply:
$this->getUser()->getPerson() is a shortcut.
https://symfony.com/blog/new-in-symfony-2-6-new-shortcut-methods-for-controllers

Reply:
I fixed it with $this->getUser()->getUserId()